server/src/main/java/org/elasticsearch/index/mapper/PostParseExecutor.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import java.io.IOException;
+
+/**
+ * Runs after a document has been parsed
+ */
+public interface PostParseExecutor {
+
+    void execute(PostParseContext context) throws IOException;
+
+    void onError(PostParseContext context, Exception e) throws IOException;
```

Comment:
does the onError part change for each field type or could it be shared? I was assuming that it's the same, hence it does not even need to be made part of this interface, potentially?

Reply:
It needs to know about the `on_script_error` behaviour.  It might be simpler to merge that into the `execute` method though, I'll try it and see if it simplifies things.